### DISCUSSION TITLE
$nuxt undefined when moving routes

### DISCUSSION THREAD

**Discussion by surfer77** [nuxt/nuxt#18087] [Questions]
I have been having issues with $nuxt undefined when browsing different routes if I had dynamic meta tags set. I added `  <NuxtPage :key="$route.fullPath" />` on `app.vue` and that solved the problem for a while. But today again I can't move between routes that load the meta tags from the api call. 

Was something changed? did   `<NuxtPage :key="$route.fullPath" />` stop working?

**Comment by surfer77** [comment]
I see in the past 3 days a lot of commits have been done regarding the router, I think it broke something.

**Reply by danielroe** [reply]
If you can reproduce a bug, please do create an issue and I'll happily take a look.

**Reply by surfer77** [reply]
The problem comes from installing pinia and @pinia/nuxt, when I uninstall those packages everything works again. 

Do you maybe have examples on how to get pinia to work? or composables + useState is what we should use?

**Reply by maxdzin** [reply]
@surfer77 For Nuxt v3 app I've used just `@pinia/nuxt` without `pinia` additionally installed as a dependency and it works like a charm.
I think it is worth trying to remove `pinia` from dependencies, but keep `@pinia/nuxt`, remove package-lock.json and node_modules and try to install again. Maybe it will help.
